playground/infrastructure/helper.py

```diff
@@ -35,37 +44,164 @@ def find_examples(work_dir: str) -> List[Example]:
+
+    If file contains tag, returns tag as a map.
+    If file contains tag but tag has incorrect format, raise error
+    If file doesn't contain tag, returns None
+
+    Args:
+        filepath: path of the file
+    """
+    with open(filepath) as parsed_file:
+        content = parsed_file.read()
+    index_of_licence_end = content.find(END_OF_LICENCE) + len(END_OF_LICENCE)
+    index_of_import_start = content.find(START_OF_IMPORT)
+    content = content[index_of_licence_end:index_of_import_start]
+
+    index_of_tag_start = content.find(BEAM_PLAYGROUND)
+    if index_of_tag_start < 0:
+        return None
+    index_of_from_start = content.find(START_OF_FROM)
+    if index_of_from_start > 0:
+        content = content[index_of_tag_start:index_of_from_start]
+    else:
+        content = content[index_of_tag_start:]
+    yaml_tag = content.replace("//", "").replace("#", "")
+    try:
+        object_meta = yaml.load(yaml_tag, Loader=yaml.SafeLoader)
+        return object_meta[BEAM_PLAYGROUND]
+    except Exception as exp:
+        print(exp)  ## todo add logErr
+        raise ValueError("found tag is not correct: " + exp.__str__())
+
+
+def _get_example(filepath: str, filename: str) -> Example:
+    """ Return an Example by filepath and filename.
+
+    Args:
+         filepath: path of the example's file.
+         filename: name of the example's file.
+
+    Returns:
+        Return an Example.
+    """
+    name = _get_name(filename)
+    sdk = _get_sdk(filename)
+    with open(filepath) as parsed_file:
+        content = parsed_file.read()
+
+    return Example(name, "", sdk, filepath, content, "", STATUS_UNSPECIFIED)
+
+
+def _match_pattern(filepath: str) -> bool:
+    """Check file to matching
+
+    Check that file has the correct extension and contains the beam-playground tag.
+
+    Args:
+        filepath: path to the file.
+
+    Returns:
+        True if file matched. False if not
+    """
+    extension = filepath.split(os.extsep)[-1]
+    if extension in SUPPORTED_SDK:
+        tag = get_tag(filepath)
+        if tag:
+            _validate(tag)
+            return True
+    return False
+
+
+def _validate(tag: dict):
+    """Validate all tag's fields
+
+    If some of the fields has incorrect format, raise error
+
+    Args:
+        tag: beam tag to validate
+    """
+    if tag.get("name") is None:
+        raise ValueError("tag doesn't contain name field: " + tag.__str__())
+    if tag.get("description") is None:
+        raise ValueError("tag doesn't contain description field: " + tag.__str__())
+    if tag.get("multifile") is None:
+        raise ValueError("tag doesn't contain multifile field: " + tag.__str__())
+    multifile = tag.get("multifile")
+    if str(multifile).lower() not in ["true", "false"]:
+        raise ValueError("tag's field multifile is incorrect: " + tag.__str__())
+    if tag.get("categories") is None:
+        raise ValueError("tag doesn't contain categories field: " + tag.__str__())
```

Comment:
Let's create a separate `.yaml` file with the list of all categories, as it is described in TDD and then validate that provided categories are in our list. It is okay if we miss some categories for now, this list will be editable and if some category is missing, there will be a possibility to add it via the same PR.

Reply:
Added.